### PR TITLE
API1000 - storage pools

### PR DESCRIPTION
**Pull request by madhav-bharadwaj**
### Description
API 1000 Support for Storage pools

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for go 1.11.x + gofmt checks .
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [X] New endpoints supported are updated in the endpoints-support.md file.
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: oneview/data_source_storage_pool.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"device_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"capacity_limit": {
+							Type:     schema.TypeString,
+							Required: true,
```
Comment (AnushaY1916) [inline oneview/data_source_storage_pool.go]: Remove required and make it as computed

File: oneview/data_source_storage_pool.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+						},
+						"device_speed": {
+							Type:     schema.TypeString,
+							Required: true,
```
Comment (AnushaY1916) [inline oneview/data_source_storage_pool.go]: Same as above

File: oneview/resource_storage_pool.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package oneview
+
+import (
+	//	"fmt"
```
Comment (jyothis-gireesan) [inline oneview/resource_storage_pool.go]: remove commented

Comment (madhav-bharadwaj) [reply oneview/resource_storage_pool.go]: Done.

**Pull request merged**